### DISCUSSION TITLE
How to Convert the Multiple images to the Video.

### DISCUSSION THREAD

**Discussion by SohamTilekar** [python-ffmpegio/python-ffmpegio#19] [Q&A] [answered]
How to Convert the Multiple images to the Video Using the ffmpegio module While tracking the progress.

I Am Asking this question because the documentation is too much confusing.

**Comment by SohamTilekar** [comment]
And what does this do.
:param progress: progress monitor object, defaults to None

**Comment by tikuma-lsuhsc** [comment, marked as answer]
@SohamTilekar - First of all, I moved the issue over to the discussion as you're asking a howto question rather than reporting an issue/bug (aside from poor documentation).

What is the exact FFmpeg command that you wish to execute with `ffmpegio`?

Let's say your command looks like this one from the [official wiki](https://trac.ffmpeg.org/wiki/Slideshow#Globpattern):

```
ffmpeg -framerate 10 -pattern_type glob -i '*.jpg' -c:v libx264 -pix_fmt yuv420p out.mp4
```

This FFmpeg command translates to [`ffmpegio.transcode`](https://python-ffmpegio.github.io/python-ffmpegio/basicio.html#ffmpegio.transcode) call as follows

```python
import ffmpegio as ff

# progress callback template (straight out of the readme example
def progress(info, done):
    pprint(info) # bunch of stats
    if done:
        print('video decoding completed')
    else:
        return check_cancel_command(): # return True to kill immediately

ff.transcode('*.jpg', 'out.mp4', 
                    framerate_in=10, 
                    pattern_type_in='glob', 
                    **{'c:v': 'libx264'}, 
                    pix_fmt ='yuv420p', 
                    progress=progress)
```

Note that:

- An input option must be given with an appended `_in` to its key name.
- If FFmpeg option name cannot be used directly as Python function keyword, use dict expansion, e.g., `**{'c:v': 'libx264'}`
